src/parsers/qasm3_parser/passes/ConstEvalPass.cpp

```diff
@@ -333,8 +333,7 @@ std::optional<ConstEvalValue> ConstEvalPass::visitBinaryExpression(
     rhsVal->type = ConstEvalValue::Type::ConstInt;
   } else if (lhsVal->type == ConstEvalValue::Type::ConstUint &&
              rhsVal->type == ConstEvalValue::Type::ConstFloat) {
-    lhsVal->value =
-        static_cast<double>(std::get<0>(lhsVal->value));
+    lhsVal->value = static_cast<double>(std::get<0>(lhsVal->value));
```

Comment:
I have no idea how that warning got here, but there seem to be two identical `else if` branches here. 
One of them can simply be removed.